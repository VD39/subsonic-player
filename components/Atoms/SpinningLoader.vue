<script setup lang="ts">
withDefaults(
  defineProps<{
    size?: number;
  }>(),
  {
    size: ICON_SIZE.medium,
  },
);
</script>

<template>
  <span :class="['centerItems', $style.spinningLoader]">
    <component
      :is="ICONS.spinner"
      ref="spinnerIcon"
      :size="size"
      :class="$style.icon"
    />

    <span class="visuallyHidden">Loading...</span>
  </span>
</template>

<style module>
.spinningLoader {
  flex-shrink: 0;
}

.icon {
  animation: spinner-loader 0.75s linear infinite;
}

@keyframes spinner-loader {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
</style>
