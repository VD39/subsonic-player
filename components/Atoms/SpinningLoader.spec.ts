import type { VueWrapper } from '@vue/test-utils';

import { mount } from '@vue/test-utils';

import SpinningLoader from './SpinningLoader.vue';

function factory(props = {}) {
  return mount(SpinningLoader, {
    props: {
      ...props,
    },
  });
}

describe('SpinningLoader', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when size prop is not set', () => {
    it('sets the default size prop to PhCircleNotch component', () => {
      expect(
        wrapper.findComponent({ ref: 'phCircleNotch' }).attributes('size'),
      ).toBe(ICON_SIZE.medium.toString());
    });
  });

  describe('when fullWidth prop is set to true', () => {
    beforeEach(() => {
      wrapper = factory({
        size: ICON_SIZE.large,
      });
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('sets the default size prop to PhCircleNotch component', () => {
      expect(
        wrapper.findComponent({ ref: 'phCircleNotch' }).attributes('size'),
      ).toBe(ICON_SIZE.large.toString());
    });
  });
});
