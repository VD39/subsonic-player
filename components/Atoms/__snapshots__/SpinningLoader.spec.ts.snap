// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpinningLoader > matches the snapshot 1`] = `"<span class="centerItems spinningLoader"><phcirclenotch class="icon" size="24"></phcirclenotch><span class="visuallyHidden">Loading...</span></span>"`;

exports[`SpinningLoader > when fullWidth prop is set to true > matches the snapshot 1`] = `"<span class="centerItems spinningLoader"><phcirclenotch class="icon" size="32"></phcirclenotch><span class="visuallyHidden">Loading...</span></span>"`;
